### ISSUE TITLE
Add `ENUM.value()` and `ENUM.key()`

### ISSUE THREAD

**Issue by Amatrelan** [godotengine/godot-proposals#3616]
### Describe the project you are working on

2.5D RPG mobile game

### Describe the problem or limitation you are having in your project

Currently when iterating over enum it returns for some reason `key` of that enum.
```gdscript
enum Enum {
  ONE,
  TWO,
}

func _ready():
  for each in Enum:
    print(each)
```

Prints out 
```txt
ONE
TWO
```

Which is OK but not exactly expected because usually this would print value of that enum key, in this case it would be
```txt
0
1
```

This caused me to believe that if I don't specify starting value it would also be string in case of, but seems to be not.
```gdscript
func _ready():
  print(Enum.ONE)
```
```txt
0
```

This causes problem if I want to use Enum as Dictionary key because it's not consistent how it works. 
Here is example how I would like to use:

```gdscript

enum Enum {
  ONE,
  TWO
}

var _dict: {}

func _ready():
  for each in Enum:
    _dict[each] = [] # <- in here it's string

func get_value(key: Enum):
  return _dict[key] # <- in here it's int
```

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add user to possibility to call in enum variable `value()` and `key()` so user can specify which one they need / want to use.

- `ENUM.key()` would return key as String.
- `ENUM.value()` would return value as Int

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Proposal is really simple and gives user better usage power with Enums.

```gdscript
enum Enum {
  ONE,
  TWO
}

var _dict: {}

func _ready():
  for each in Enum:
    _dict[each.key()] = [] 

func get_value(key: Enum):
  return _dict[key.key()]
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I would prefer to use this in all dictionaries that I use, creates much less runtime bugs.
But there is possibility to work around this to create for function for every enum as:
```gdscript
enum Enum {
  ONE,
  TWO
}

func get_enum_as_str(value: Enum):
  var iter = 0
  var output = ""
  for each in Enum:
    if iter == value:
      return each
    iter += 1
```
This would be needed to create to every enum you use in Dictionary in game.

EDIT: Found out actually there is some finicky way to get it with
```gdscript
func get_value(key: Enum):
  return _dict.keys()[key]
```
So workaround is shorter, but that's not clear to developer that they need to workaround like this.

### Is there a reason why this should be core and not an add-on in the asset library?

GDScript cannot be modified outside core and there is no way to pass enum as whole to function which causes that I would need to create workaround for each enum.

**Comment by Calinou** [comment]
Enums are dictionaries in `3.x`, but they were changed in `master` as part of the GDScript rewrite.

**Comment by aaronfranke** [comment]
While enums are usually represented as integers internally, this is an implementation detail. Interacting with the integer values behind the enum is not the "ideal" way to use enums. If you want to get the backing integer of `Enum.ONE`, I think it should be required to do `int(Enum.ONE)` etc.

**Comment by Amatrelan** [comment]
> Enums are syntax sugar for dictionaries in `3.x`, but they were changed in `master` as part of the GDScript rewrite.

Yeah thought so when I when I came to bug in it, and it printed integers then godotengine/godot#55491 

> While enums are usually represented as integers internally, this is an implementation detail. Interacting with the integer values behind the enum is not the "ideal" way to use enums. If you want to get the backing integer of `Enum.ONE`, I think it should be required to do `int(Enum.ONE)` etc.

Yeah I agree it's not ideal but I don't have used any language except one that don't return `Enum.ONE` as integer and that's why I think it's expected behavior how enums work. But as I proposed there should be simple way to user to say which value user wants out.

As for `int(Enum.ONE)` I don't agree, if default behavior would be returning `"ONE"`, because this would basically make it so that `int("ONE")` should return same value, which is not true.

This is why I want explicit functions to get key or value out from enum.
